### DISCUSSION TITLE
Recommended approach for 3d scene + satelite map

### DISCUSSION THREAD

**Discussion by h0uter** [zauberzeug/nicegui#939] [Q&A] [answered]
### Question

Hey all,

For our robotics project we would like to use Nicegui to visualize 3D navigation graphs together with "google maps" style satellite images. What is the best approach for this?

I have seen the [leaflet example for creating a custom map element](https://github.com/zauberzeug/nicegui/tree/main/examples/map) and we are already using the [standard 3D scene element](https://nicegui.io/documentation/scene).
The question is what the recommended approach is for combining both?

**Comment by falkoschindler** [comment, marked as answer]
Interesting question. I think I would use something like the Google Map Tiles API and NiceGUI's 3D textures:

```py
url = 'https://www.google.de/maps/vt/pb=!1m5!1m4!1i11!2i2134!3i1353!4i128!2m2!1e1!3i946!3m9!2sen!3sde!5e1105!12m1!1e4!12m1!1e47!12m1!1e3!4e0!5m2!1e0!5f2!23i10203575!23i1381033!23i1368782!23i1368785!23i47025228!23i4592408!23i4640515!23i4897086!23i4940989!23i1375050!23i4536287!23i47054629'
with ui.scene() as scene:
    scene.texture(url, [[[-2, 2, 0], [2, 2, 0]],
                        [[-2, -2, 0], [2, -2, 0]]])
```

Note that I don't have an API key at hand and therefore simply grabbed an URL from the network traffic on Google Maps for this proof of concept.
